presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -742,6 +743,25 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<A
         }
 
         @Override
+        protected Type visitAtTimeZone(AtTimeZone node, StackableAstVisitorContext<AnalysisContext> context)
+        {
+            Type valueType = process(node.getValueExpression(), context);
+            process(node.getTimeZoneSpecifier(), context);
+            if (!valueType.equals(TIME_WITH_TIME_ZONE) && !valueType.equals(TIMESTAMP_WITH_TIME_ZONE)) {
+                throw new SemanticException(TYPE_MISMATCH, node.getValueExpression(), "Type of value must be TIME_WITH_TIME_ZONE or TIMESTAMP_WITH_TIME_ZONE (actual %s)", valueType);
+            }
+            Expression timeZoneSpecifier = node.getTimeZoneSpecifier();
+            if (!(timeZoneSpecifier instanceof IntervalLiteral) && !(timeZoneSpecifier instanceof StringLiteral)) {
```

Comment:
not to clear on the spec here, but should we be checking for the Type here instance of instanceof? Or does it strictly have to be one of these two literals?